tests/migration/openldap_to_389ds.pm

```diff
@@ -0,0 +1,105 @@
+# Copyright (C) 2021 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: The openldap_to_ds tool will attempt to automatically migrate
+# custom schema, backens, some overlays from openldap to 389ds.
+#
+# Maintainer: wegao <wegao@suse.com>
+
+use base 'consoletest';
+use testapi;
+use strict;
+use warnings;
+use utils;
+use lockapi;
+use Utils::Systemd qw(systemctl disable_and_stop_service);
+use registration qw(add_suseconnect_product);
+use version_utils;
+
+sub run {
+    my ($self) = @_;
+    my $password = $testapi::password;
+    $self->select_serial_terminal;
+
+    # Install 389-ds and sssd on client
+    zypper_call("in 389-ds sssd sssd-tools");
+    zypper_call('info 389-ds');
+    # Install openldap since we need use slaptest tools
+    add_suseconnect_product('sle-module-legacy') if is_sle;
+    zypper_call("in openldap2 sssd-ldap openldap2-client");
+    # Install glibc for Tumbleweed
+    zypper_call("in glibc") if is_tumbleweed;
```

Comment:
Why would Tumbleweed need to be handled different here?

Our tests should always consider openSUSE Tumbleweed as the default product without a version. That means any differing behaviour should have an explicit exclude rule for the "older products" and potentially an explanation why the behaviour should differ. You can try to negate the logic check.

See https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md#coding-style for more details

Reply:
I am debugging on this now, since i encounter no /etc/nsswitch.conf error when i running tumbleweed(So i need try add extra package in tumbleweed).  I can set this PR to WIP currently.  Will notify you once i am ready.

Reply:
> I am debugging on this now, since i encounter no /etc/nsswitch.conf error when i running tumbleweed(So i need try add extra package in tumbleweed). I can set this PR to WIP currently. Will notify you once i am ready.

I think that in TW you will find the system files in: /usr/etc/nsswitch.conf, https://bugzilla.opensuse.org/show_bug.cgi?id=1175238#c10

So you will need to create the file if doesn't exit, or copy it from usr/etc if we're on sle

Reply:
@okurz @foursixnine  TW also pass now  https://openqa.opensuse.org/tests/1663905#
could you help have a check again?

Reply:
Still, treating Tumbleweed as something special is wrong here. As written in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12075/files#r590125044 Tumbleweed just happens to have more recent versions of packages but again: Keep a version check, but not use the "always-rolling" Tumbleweed for a version check, use a negated version check, e.g.

```suggestion
    zypper_call 'in glibc' unless is_sle('<15-sp3') or is_leap('<15.3');
```

Reply:
Sorry forget remove this debug info. We do not need this anymore. 
also with latest verify run:
https://openqa.opensuse.org/t1668418